src/Mod/Draft/draftviewproviders/view_layer.py

```diff
@@ -216,14 +216,43 @@ def loads(self, state):
 
     def updateData(self, obj, prop):
         """Execute when a property from the Proxy class is changed."""
-        if prop == "Group":
-            for _prop in ("LineColor", "ShapeAppearance", "LineWidth",
-                          "DrawStyle", "Visibility"):
-                self.onChanged(obj.ViewObject, _prop)
+        if prop == "Label":
+            self._paint_tree_icon(obj.ViewObject)
 
-    def change_view_properties(self, vobj, prop):
-        """Iterate over the contents and change the properties."""
-        obj = vobj.Object
+    def change_view_properties(self, vobj, prop, old_prop=None, targets=None):
+        """Change the properties of the targets, or of all objects in the layer group.
+
+        A child's property is only changed if it is not overridden (if its value
+        matches old_prop).
+        """
+
+        def _color_is_same(col1, col2):
+            # Ignore alpha and round RGB values
+            return [round(c, 2) for c in col1[:3]] == [round(c, 2) for c in col2[:3]]
+
+        def _app_material_is_same(mat1, mat2):
+            for prop in (
+                "AmbientColor",
+                "DiffuseColor",
+                "EmissiveColor",
+                "Shininess",
+                "SpecularColor",
+                "Transparency",
+            ):
+                if "Color" in prop:
+                    if not _color_is_same(getattr(mat1, prop), getattr(mat2, prop)):
+                        return False
+                elif getattr(mat1, prop) != getattr(mat2, prop):
+                    return False
+            return True
+
+        def _prop_is_same(prop1, prop2):
+            if type(prop1) == tuple:
```

Comment:
I think the linter's right here, I was taught to use `isinstance()` for this use-case (here and below).